libpod/oci_linux.go

```diff
@@ -405,6 +405,14 @@ func (r *OCIRuntime) stopContainer(ctr *Container, timeout uint) error {
 		stopSignal = uint(syscall.SIGTERM)
 	}
 
+	defer func() {
```

Comment:
This is not needed or desired. The cleanup process should take care of it.

Reply:
In the case of restart, the previous network namespaces are not getting cleaned up. I am slightly baffled as to where the removal of the netns bind mount is  occurring in the `stop` call in general. It is probably horribly obvious, but I see directly after the call to [`c.StopWithTimeout`](https://github.com/containers/libpod/blob/1bbcb2fc56c1655269930c83628ef5f5802f1887/pkg/adapter/containers.go#L100) the bind mount to the netns is gone `/run/user/1000/netns/...`. However, when I throw a debug statement at the end of the `c.StopWithTimeout` function it still exists. 
I'll continue to look for the appropriate place to get this cleaned up.

Reply:
`stop` itself does not remove them. On the container stopping, the cleanup process (`podman container cleanup`, spawned by conmon) will fire, removing the network namespace, mounts, etc. On restart, we've just been reusing the old network namespace, instead of configuring a new one.

Reply:
With the new flow, Podman is now creating the network namespace instead of the runtime, and it is being bind mounted so that it can be passed to slirp and the runtime. I'm storing this in `c.state.NetNS`, which wasn't being set/used with the previous rootless flow. So, there needs to be an extra step of explicitly cleaning up the podman created and persisted NetNS. I traced the `podman container cleanup` call, specifically the `CleanupContainers` function in the `pkg/adapter/containers.go` file, and I don't see it being called in either model with a `restart`  cmd, so I'm not entirely sure how to how to move forward.

Reply:
Does restart need to clean up the old namespace? We've been reusing them for containers running as root.